cmd/node/list.go

```diff
@@ -53,20 +56,24 @@ func newListCommand(vp *viper.Viper) *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringVarP(
+	formattingFlags := pflag.NewFlagSet("Formatting", pflag.ContinueOnError)
+	formattingFlags.StringVarP(
 		&listOpts.output, "output", "o", "table",
 		`Specify the output format, one of:
  json:     JSON encoding
  table:    Tab-aligned columns
  wide:     Tab-aligned columns with additional information`)
-	cmd.RegisterFlagCompletionFunc("output", func(_ *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
+	listCmd.RegisterFlagCompletionFunc("output", func(_ *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
 		return []string{
 			"json",
 			"table",
 			"wide",
 		}, cobra.ShellCompDirectiveDefault
 	})
-	return cmd
+	listCmd.Flags().AddFlagSet(formattingFlags)
+
+	listCmd.SetUsageTemplate(template.Usage(formattingFlags, config.ServerFlags))
```

Comment:
We may want to look into wrapping this in some other function that auto-adds relevant global flags, if that makes sense. I'd rather pass in the `*cmd.Command` into the usage, than pass `template.Usage` into `cmd` if that makes sense.

In that function we can look at the name of the command and take some appropriate steps perhaps?

Reply:
This is what I wanted to do initially but it turns out that [`AddFlagSet(newSet *FlagSet)`](https://godoc.org/github.com/spf13/pflag#FlagSet.AddFlagSet) flattens the provided `FlagSet`. This means that one can never recover the different flags sets (and name) once they're added to a command (which is rather unfortunate) given the `*cmd.Command` only.

Reply:
I think you may have misunderstood what I was proposing. I wrote up some pseudo code to try to express what I think is possible (excuse the rough syntax please):

```go
// common file

var ServerFlags ...

func Usage() {
	// ...
}

func AddUsage(cmd*cobra.Command, flags ...) {
	var extra []flag

	// some command-specific flags
	if cmd.Name() == "observe" {
		extra = append(extra, ServeFlags)
	}

	extra = append(extra, ConfigFlags)
	cmd.SetUsageTemplate(Usage(flags..., extra))
}

// ------------------------------------

// command file

func NewCommand() {
	cmd := &cobra.Command{...}

	set1 := cmd.Flags()...
	set2 := cmd.Flags()...

	common.AddUsage(cmd, set1, set2)
}
```

I think the main benefit is that `config.ServerFlags` gets auto-added to all the relevant commands in the same place.

Reply:
I indeed misunderstood. This is a very nice idea, will implement!

Reply:
@glibsm I gave it a go but it turns out to be problematic for sub-subcommands (e.g. `hubble nodes list`). At creation time, such a sub-subcommand is not attached to a parent command and it's name is simply `list` which is not unique enough to identify. I've tried several variants of the above but in the end, the current implementations is easier to use/understand so I'll leave it at that.

Reply:
Thanks for giving it a shot and proving the update.